src/pages/sym.mdx

```diff
@@ -0,0 +1,132 @@
+
+##### Location
+
+The location shall be of the form `<bank>:<address>`, both `<bank>` and `<address>` being case-insensitive hexadecimal numbers, possibly zero-padded, separated by a colon (`:`, U+003A).
+Typical hexadecimal prefixes (`0x`, `$`, etc.) are not allowed.
+An implementation may also allow the bank to be the case-insensitive keyword `BOOT` to specify symbols in the loaded boot ROM.
+If the bank is omitted, then the colon must be as well; the results of such lines are implementation-defined.
+
+The address must be in 16-bit range (0 to $FFFF inclusive), the bank must be in 32-bit range (0 to $FFFFFFFF inclusive); implementations may abandon processing the line if either number is larger than these limits.
+Implementations may accept bank numbers higher than $FFFFFFFF.
+Nonsensical locations (e.g. `2:9800`) may produce a warning.
+
+Two locations are considered identical if they have the same bank and the same address.
+Two distinct symbols can have identical locations; in particular, <LatestStableLink to="rgbasm.5#Unions">unions</LatestStableLink> facilitate this.
+
+Locations are strictly ordered by memory region (ROM0, ROMX, VRAM, SRAM, WRAM0, WRAMX, Echo RAM, OAM, I/O, HRAM, IE), then by bank, then by address.
+Entries may or may not be sorted in the sym file; for example, RGBLINK before v0.4.0 output symbols in the same order that it read them from its input files.
+Some sym files are piped through [`sort(1)`](https://man.openbsd.org/sort.1), which often produces files sorted by bank then address, but not by region.
+
+Sym files do not carry information about the memory region a symbol belongs to, but that can be inferred from the address most of the time.
+This detection is sometimes ambiguous: for example, `0:A000` could be one past the end of VRAM, or at the beginning of SRAM.
+The following heuristics are suggested:
+- If the symbol's bank is in range of exactly one of the candidate regions, assign it to that one. (*There should be no ambiguity, e.g. `2:A000`.*)
+- If the symbol is local and is "attached" to a global symbol (see the "presentation shortening" below), assign it to the same region as the global symbol. (*Local labels should "follow" the global symbol they are attached to, if any.*)
+- If the symbol's name ends with (case-insensitive) "end", assign it to the earlier region. (*This is a common naming convention, and "one-past-end" labels are typically "end markers".*)
+- If all else fails, assign it to the later region (e.g. `0:A000` would be assigned to RAM). (*This is correct more often than not in practice.*)
+
+##### Name
+
+The symbol name can contain any characters except ASCII spaces, tabs, non-printable ASCII characters, or characters excluded by earlier processing (e.g. semicolons).
```

Comment:
What about Unicode garbage? I'd be explicit in the ranges supported.
(And don't even attempt to find a reasonable subset of Unicode. The Unicode Consortium will ruin your work almost instantly as soon as they add a random character you don't expect.)

Reply:
The file format is specified to be "well-formed UTF-8", so that's already excluded.

Reply:
I didn't mean invalid sequences (like a 0xc0), but rather lots of Unicode things that aren't really printable.

Reply:
I'm not sure what you want, given that you both say "I'd be explicit in what's supported" and "don't try to be explicit because updates will bite you".

Reply:
My point is that you basically have three ways to handle Unicode characters that will survive an update cycle: accept them all as identifiers (essentially treating them as letters), reject them all, or have an explicit, non-changing whitelist/blacklist (for instance, JSON only accepts a few explicitly designated codepoints as whitespace).

While your current wording would imply the former (Unicode whitespace is a valid symbol character), I'm not too sure if this is what was intended.

Reply:
I'm not sure what the "update cycle" is, but my intention is that such characters should be treated as part of identifiers; whether they can be processed and how is left up to the implementation. ASCII is already encouraged as per the above, so I think that's reasonable.

Reply:
In that case I'd make it explicit that excluded characters are codepoints U+0000 through U+0020. (You can add a note indicating what those codepoints are, but it's just C0 controls (including tabs) and your good old common space.)

Reply:
Well, that and U+007F.

Reply:
I think you just proved why an explicit list is important :P